connection-service/new-admin-vm/Install-ConnectionServer.ps1

```diff
@@ -877,122 +877,116 @@ brokerIpaddress=$ipaddressString
-			}
-		}
-
-
-		Script Set_CAM_Envionment_And_Reboot
-		{
-		    # depends on both services being installed to ensure the reboot at the end will start both services properly.
-			DependsOn  = @("[Script]Install_Auth_file", "[Script]Install_Broker")
-			GetScript  = { @{ Result = "Set_CAM_Envionment_And_Reboot" } }
+                Set-Content $cbPropertiesFile $cbProperties
+                Set-Content $cbHomePropertiesFile $cbProperties
+                Write-Host "Broker configuration file generated."
+
+                #----- setup security trust for LDAP certificate from DC -----
+
+                #second, get the certificate file
+                $issuerCertFileName = "issuercert.cert"
+                Write-Host "Looking for Issuer certificate for $dcvmfqdn"
+
+                $foundCert = $false
+                $caCert = $null
+                $loopCountRemaining = 180
+
+                # LDAPS Port and Host
+                $port=636
+                $hostname=$using:dcvmfqdn
+                #loop until it's created
+                while(-not $foundCert)
+                {
+                    $cert = $null
+                    try {
+                        Write-Host "Looking for LDAPS certificate for $hostname"
+                        $tcpclient = new-object System.Net.Sockets.tcpclient
+                        $tcpclient.Connect($hostname,$port)
+
+                        #Authenticate with SSL
+                        $sslstream = new-object System.Net.Security.SslStream -ArgumentList $tcpclient.GetStream(),$false
+
+                        $sslstream.AuthenticateAsClient($hostname)
+                        $cert =  [System.Security.Cryptography.X509Certificates.X509Certificate2]($sslstream.remotecertificate)
+                        Write-Host "Found Certificate for $hostname, looking for Issuer Certificate"
+
+                        $chain = New-Object -TypeName System.Security.Cryptography.X509Certificates.X509Chain
+                        # Build the certificate chain from the file certificate
+                        $chain.Build($cert)                        
+                        $listOfCertificates = ($chain.ChainElements | ForEach-Object {$_.Certificate})
+                        
+                        #last one is the root in chain
+                        $caCert=$listOfCertificates[-1]
+                        $content = @(
+                            '-----BEGIN CERTIFICATE-----'
+                            [System.Convert]::ToBase64String($caCert.RawData, 'InsertLineBreaks')
+                            '-----END CERTIFICATE-----'
+                        )
+                       
+                        $content | Out-File -FilePath "$env:systemdrive\$issuerCertFileName" -Encoding ascii
+                        $foundCert=$true
+                    } catch {
+                        throw "Failed to retrieve issuer certificate from $hostname`:$port because $_"
```

Comment:
@cptera - I don't think we want to re-throw here until the retires are used up. (Or some other similar restructuring that makes sense :)